oauthapplication/client.go

```diff
@@ -0,0 +1,116 @@
+// Package oauthapplication provides the OAuth applications API.
+package oauthapplication
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+
+	"github.com/clerk/clerk-sdk-go/v2"
+)
+
+//go:generate go run ../cmd/gen/main.go
+
+const path = "/oauth_applications"
+
+type Client struct {
+	Backend clerk.Backend
+}
+
+func NewClient(config *clerk.ClientConfig) *Client {
+	return &Client{
+		Backend: clerk.NewBackend(&config.BackendConfig),
+	}
+}
+
+// fetches a single OAuth application by its ID.
+func (c *Client) Get(ctx context.Context, id string) (*clerk.OAuthApplication, error) {
+	path, err := clerk.JoinPath(path, id)
+	if err != nil {
+		return nil, err
+	}
+	req := clerk.NewAPIRequest(http.MethodGet, path)
+	resource := &clerk.OAuthApplication{}
+	err = c.Backend.Call(ctx, req, resource)
+	return resource, err
+}
+
+type ListParams struct {
+	clerk.APIParams
+	clerk.ListParams
+}
+
+func (params *ListParams) ToQuery() url.Values {
+	q := params.ListParams.ToQuery()
+	return q
+}
+
+// Retrieves all OAuth applications.
+func (c *Client) List(ctx context.Context, params *ListParams) (*clerk.OAuthApplicationList, error) {
+	req := clerk.NewAPIRequest(http.MethodGet, path)
+	req.SetParams(params)
+	list := &clerk.OAuthApplicationList{}
+	err := c.Backend.Call(ctx, req, list)
+	return list, err
+}
+
+type OAuthApplicationCreateParams struct {
+	clerk.APIParams
+	Name        string `json:"name"`
+	CallbackURL string `json:"callback_url"`
+	Scopes      string `json:"scopes"`
+	Public      bool   `json:"public"`
+}
+
+// Creates a new OAuth application with the given parameters.
+func (c *Client) Create(ctx context.Context, params *OAuthApplicationCreateParams) (*clerk.OAuthApplicationWithClientSecret, error) {
```

Comment:
:question: Do we need the different type here? Why can't `Create` return a `*clerk.OAuthApplication`?

```suggestion
func (c *Client) Create(ctx context.Context, params *CreateParams) (*clerk.OAuthApplication, error) {
```